src/main/java/uk/nhs/digital/emailadapter/schedulingtasks/EmailScheduledTask.java

```diff
@@ -0,0 +1,180 @@
+      SearchFilterCollection searchFilterCollection =
+          new SearchFilter.SearchFilterCollection(LogicalOperator.And);
+      searchFilterCollection.add(
+          new SearchFilter.IsEqualTo(EmailMessageSchema.From, getParameter("EMS_REPORT_SENDER")));
+      searchFilterCollection.add(new SearchFilter.IsEqualTo(EmailMessageSchema.IsRead, false));
+      FindItemsResults<Item> findResults =
+          service.findItems(WellKnownFolderName.Inbox, searchFilterCollection, view);
+
+      while (findResults.getTotalCount() > 0) {
+        for (Object item : findResults.getItems()) {
+          EmailMessage emailMessage = (EmailMessage) item;
+          emailMessage
+              .load(new PropertySet(BasePropertySet.FirstClassProperties, ItemSchema.MimeContent));
+          log.info("attachment count: {} ", emailMessage.getAttachments().getCount());
+
+          Attachment attachment = emailMessage.getAttachments().getItems().get(0);
+          stopwatch.finishStage("Getting html attchement from email");
+
+          if (attachment instanceof FileAttachment) {
+            FileAttachment fileAttachment = (FileAttachment) attachment;
+            fileAttachment.load();
+
+            // convert bytes[] to string
+            String htmlString = new String(fileAttachment.getContent(), StandardCharsets.UTF_8);
+            byte[] transform = new PDFTransformer().transform(Jsoup.parse(htmlString).html());
+            stopwatch.finishStage("pdf transformation");
+
+            // Create an email message and set properties on the message.
+            EmailMessage message = new EmailMessage(service);
+            message.setSubject(getParameter("EMS_REPORT_SUBJECT"));
+            message.setBody(new MessageBody(getParameter("EMS_REPORT_BODY")));
+            message.getToRecipients().add(getParameter("EMS_REPORT_RECIPIENT"));
+
+            FileAttachment addFileAttachment = message.getAttachments()
+                .addFileAttachment(createFileName(Jsoup.parse(htmlString)), transform);
+            addFileAttachment.setContentType("application/pdf");
+            stopwatch.finishStage("Added pdf attchement to mail");
+
+            message.send();
+            stopwatch.finishStage("Sent an email with pdf attachement");
+
+          } else if (attachment instanceof ItemAttachment) {
+            ItemAttachment itemAttachment = (ItemAttachment) attachment;
+
+            String name = itemAttachment.getName();
+            log.info("ItemAttachment name : {} ", name);
+            Item attachedItem = itemAttachment.getItem();
+            log.info("ItemAttachment attachedItem : {} ", attachedItem);
+          }
+          emailMessage.setIsRead(true);
+          emailMessage.update(ConflictResolutionMode.AlwaysOverwrite);
+          stopwatch.finishStage("Making email unread after reading email");
+        }
+        findResults = service.findItems(WellKnownFolderName.Inbox, searchFilterCollection, view);
+      }
+    } catch (Exception e) {
```

Comment:
location of this try catch should be inside the loop. If one iteration fails, execution should continue with the other emails rather than failing completely for the whole batch.